blockchain/stake/staketx.go

```diff
@@ -878,8 +888,12 @@ func CheckSSGenVotes(tx *wire.MsgTx, isTreasuryEnabled bool) ([]TreasuryVoteTupl
 	}
 
 	// Check to make sure there aren't too many outputs.
-	if len(tx.TxOut) > MaxOutputsPerSSGen {
-		str := "SSgen tx has too many outputs"
+	maxOuts := maxOutputsPerSSGen
```

Comment:
Unfortunately, while this is technically correct, the rule itself can't be changed without gating it behind the result of a new consensus vote, because it could be used to fork nodes off the chain otherwise.  This method is used to determine if a vote is considered a vote or not which in turn determines whether or not it is allowed into the stake tree.  While this change does gate it behind the treasury vote, that has already happened and been activated and therefore the rules that apply to that vote is active can't be changed.

What I would suggest is adding a comment calling out that this technically should have done what this change is suggesting here, but that making that change requires a consensus vote.

Reply:
Good point, I'll update it to just be a comment.  I suppose the implication is that if a ticket happens to have the max number of inputs, and then the vote for that ticket also contains a treasury vote, that vote tx will get rejected (resulting in a missed vote in which case the ticket can be revoked).